backend/app/models/settings.py

```diff
@@ -30,12 +32,16 @@ class Settings(BaseSettings):
 
     @field_validator("output_file_paths", mode="before")
     @classmethod
-    def convert_to_list_of_file_paths(cls, output_file_names: list[str]) -> list[Path]:
+    def convert_to_list_of_file_paths(
+        cls, output_file_names: frozenset[str]
+    ) -> frozenset[Path]:
         """Convert the list of file names in the environment variables into a list of Path objects."""
-        return [
-            Path(__file__).parent.parent / "data" / file_name
-            for file_name in output_file_names
-        ]
+        return frozenset(
+            [
+                Path(__file__).parent.parent / "data" / file_name
```

Comment:
Nit: you may want an `absolute()` in here somewhere to make the paths more obvious.